xbmc/pvr/addons/PVRClients.cpp

```diff
@@ -324,17 +339,30 @@ bool CPVRClients::IsCreatedClient(int iClientId) const
   return GetCreatedClient(iClientId, client);
 }
 
-bool CPVRClients::IsCreatedClient(const std::string& id) const
+bool CPVRClients::IsCreatedClient(const std::string& addonId, const std::string& instanceId) const
 {
   CSingleLock lock(m_critSection);
   for (const auto& client : m_clientMap)
   {
-    if (client.second->ID() == id)
+    if (client.second->ID() == addonId && client.second->GetUsedInstanceID() == instanceId)
       return client.second->ReadyToUse();
   }
   return false;
 }
 
+bool CPVRClients::HasIndependendEditedInstanceId(const std::string& id) const
```

Comment:
What does "independent" and "edited" mean hear? Independent from what? "Edidited" by whom? Maybe just call this "HasInstanceId"? 

Or wouldn't it be better that we always have an instance id? Would make all the special cases you currently check obsolete.

Reply:
The only problem was that a fallback id is created in the instance class using class pointer, which then makes it unusable for PVR.

The name was come from function name added by `GetKnownIndependendEditedInstanceIds` in `CBinaryAddonManager`.

Problem is that by other addons the instance Id's created and handled different, as example by inputstream addon where it gives always new id's on opened streams.

Reply:
However, the two functions fly out again because they are not used.